tools/stylelint/no-ampersand-beyond-selector-start.ts

```diff
@@ -37,43 +36,37 @@ const plugin = createPlugin(ruleName, (isEnabled: boolean, _options?) => {
     }
 
     root.walkRules(rule => {
-      if (
-        rule.parent.type === 'rule' &&
-        isStandardSyntaxRule(rule) &&
-        isStandardSyntaxSelector(rule.selector) &&
-        // Using the ampersand at the beginning is fine, anything else can cause issues in themes.
-        rule.selector.indexOf('&') > 0) {
-
-        const mixinName = getClosestMixinName(rule);
-
-        // Skip rules inside private mixins.
-        if (!mixinName || !mixinName.startsWith('_')) {
-          utils.report({
-            result,
-            ruleName,
-            message: messages.expected(),
-            node: rule
-          });
-        }
+      if (rule.parent.type === 'rule' &&
+          isStandardSyntaxRule(rule) &&
+          isStandardSyntaxSelector(rule.selector) &&
+          hasInvalidAmpersandUsage(rule.selector)) {
+        utils.report({
+          result,
+          ruleName,
+          message: messages.expected(),
+          node: rule
+        });
       }
     });
   };
+});
 
-  /** Walks up the AST and finds the name of the closest mixin. */
-  function getClosestMixinName(node: Node): string | undefined {
-    let parent = node.parent;
-
-    while (parent) {
-      if (parent.type === 'atrule' && parent.name === 'mixin') {
-        return parent.params;
-      }
+function hasInvalidAmpersandUsage(selector: string): boolean {
+  let index = selector.indexOf('&');
 
-      parent = parent.parent;
+  while (index > -1) {
+    // The ampersand is invalid if it's used after the start of
+    // the selector and not in a combination with another selector.
+    if (index > 0 && selector[index - 1] === ' ' && (index === selector.length - 1 ||
+        selector[index + 1] === ' ' || selector[index + 1] === ',')) {
```

Comment:
Is it expected that `hasInvalidAmpersandUsage('whatever, .mat-select &.combined')` returns `false`? I'm not too involved with this rule, but would it be simpler to just enforce that an ampersand can _only_ be used at the beginning? (as the rule name suggests)

Reply:
I think it's only really a problem when it's something like `.foo & {`, but `&.foo` should work fine.

Reply:
Yeah, I assume I'm not too familiar with how this rule should work; but given this _seems_ expected, the logic seems good to me 👍 

My impression was that e.g. `.my-select &.foo` should result in an error in themes because it would be odd to have something like: `.my-select .my-dark-theme.foo` (unless I miss something here)

Reply:
You're right, I had the impression that Sass was smarter about how it resolves ampersands. I've pushed a change to handle it correctly.